packages/core/src/database/update-where.ts

```diff
@@ -38,10 +38,25 @@ export const buildUpdateWhere: BuildUpdateWhere = <
   const isKeyOfSchema = isKeyOf(schema);
   const connectKeyValueWithEqualSign = (data: Partial<Schema>) =>
     Object.entries(data)
-      .map(
-        ([key, value]) =>
-          isKeyOfSchema(key) && sql`${fields[key]}=${convertToPrimitiveOrSql(key, value)}`
-      )
+      .map(([key, value]) => {
+        if (!isKeyOfSchema(key)) {
+          return;
+        }
+
+        if (value && typeof value === 'object' && !Array.isArray(value)) {
```

Comment:
seems `if (typeof value === 'object' && !Array.isArray(value)) {` is enough.

Reply:
type of null is a object?